assets/javascripts/redmine_s3.js

```diff
@@ -0,0 +1,15 @@
+/*
+ * This script handles thumbnail image missing errors
+ */
+
+$(function() {
```

Comment:
I think we should generate thumbnails on upload images. Like we do this
using carrierwave or paperclip. And just show the images where we need
them. It's a bad idea that during showing images you have a fallback which
does a magic automatically. You violate SRP rule in SOLID set of rules.
Imagine that we have to push thumbnails to a possible mobile app - in this
case we have to write the workaround like this there too.
On Apr 1, 2015 6:29 PM, "redcloak" <notifications@github.com> wrote:

> In assets/javascripts/redmine_s3.js
> <https://github.com/ka8725/redmine_s3/pull/32#discussion_r27580073>:
>
> > @@ -0,0 +1,15 @@
> > +/*
> > + * This script handles thumbnail image missing errors
> > + */
> > +
> > +$(function() {
>
> So that we are not checking whether thumbnail exists on s3. We just give a
> link where thumbnail is supposedly stored without any S3 api calls, and if
> browser encounters an error (because it can't load the image) - then we
> make an Ajax call and redmine creates a thumbnail for us.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ka8725/redmine_s3/pull/32/files#r27580073>.
>